src/AppTokenRetrival.tsx

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react'
+
+import { Mount } from 'react-lifecycle-components'
+import { apolloClient } from './client/apolloClient'
+import { StorageContainer } from './utils/StorageContainer'
+
+export const AppTokenRetrival = ({ children }) => (
+  <StorageContainer>
+    {(storageState) => (
+      <Mount
+        on={() => {
+          if (location.hash.includes('token=')) {
+            const token = location.hash.replace('#token=', '')
+            console.log('setting token to', token)
+            apolloClient!.subscriptionClient.close(true, true)
+            storageState.setToken(token)
```

Comment:
shouldn't this happen in the reverse order?

Reply:
hmm, we do in this order in other places 🤔

Reply:
but it makes more sense to set the token first, and that worked, so doing that instead 👨‍🔬